nevergrad/optimization/base.py

```diff
@@ -391,33 +391,36 @@ def ask(self, cheap_constraints_checker: Any = None) -> Candidate:
             The candidate to try on the objective function. Candidates have field `args` and `kwargs` which can be directly used
             on the function (`objective_function(*candidate.args, **candidate.kwargs)`).
         """
-        num_tries = 0
-        while True:  # Until we satisfy cheap constraints.
+        for callback in self._callbacks.get("ask", []):
+            callback(self)
+        MAX_TENTATIVES = 1000
+        current_num_ask = self.num_ask
+        for k in range(MAX_TENTATIVES):
             # call callbacks for logging etc...
-            for callback in self._callbacks.get("ask", []):
-                callback(self)
+            is_suggestion = False
```

Comment:
this all method is veeery sensitive...  some things are a bit tricky (like the fact that suggestions are  not registered will others are)